content/client-lib-development-guide/features.textile

```diff
@@ -682,18 +693,17 @@ h4. ChannelOptions
 * @(TB1)@ options provided when instancing a channel, see "Java ChannelOptions":https://github.com/ably/ably-java/blob/master/src/io/ably/types/ChannelOptions.java as a reference
 * @(TB2)@ The attributes of @ChannelOptions@ consist of:
 ** @(TB2a)@ @encrypted@ boolean - when true, enables automatic encryption & decryption of all messages
-** @(TB2b)@ @cipherParams@ @CipherParams@ - when encrypted is true, the cipher params are required
+** @(TB2b)@ @cipherParams@ - This is required if @encrypted@ is true. It consists of either:
+*** @(TB2b1)@ A @CipherParams@ instance, or
+*** @(TB2b2)@ an options hash (or language equivalent) consisting of the key @key@ with value a key. In this case, the client library should call `getDefaultParams`, passing it the key, to obtain a @CipherParams@ instance
```

Comment:
I agree that the `ChannelOptions.encrypted` is redundant.

> I question the value of further simplifying it so that you specify the key only as a channel option, because it doesn't really scale well to passing other non-default cipher params.

If you need more params, you can pass them as a normal `CipherParams`, or a hashmap in dynamic languages. This would just provide a straightforward way to do the most common thing, which is just taking a key. In JavaScript and friends this gives little advantage, because making a hashmap is very convenient and idiomatic, but e. g. in iOS having to construct a CipherParams would lead to something like:

```objective-c
[channels get:@"foo" options:[[ARTChannelOptions alloc] initWithCipher: [ARTCrypto getDefaultParams:@"key"]]]
```
```java
// Java equivalent.
channels.get("foo", new ChannelOptions(Crypto.getDefaultParams("key")));
```

What I'm proposing is providing this as a convenience:

```objective-c
[channels get:@"foo" options:[ARTChannelOptions optionsWithCipherKey:@"key"]]
```
```java
// Java equivalent.
channels.get("foo", ChannelOptions.withCipherKey("key"));
```

Or even just:

```objective-c
[channels get:@"foo" encryptedWithKey:@"key"];
```
```java
// Java equivalent.
channels.get("foo", "key");
```

> Calling it cipherParams as opposed to cipher makes it sound like you're configuring it but not necessarily enabling it on the channel unless encrypted: true is included. And we can still specify that the type expected is a CipherParams object.

I don't have a strong opinion but I tend to think that just `cipher` sounds better.